File: ruby/bob/bob.rb
class Bob
  def hey(sentence)
    return 'Whoa, chill out!' if sentence.strip.end_with?('!') || sentence.strip =~ /[A-Z ]{5,}+/ && !sentence.include?("Let's ")
    return 'Sure.' if sentence.end_with? '?'
    return 'Fine. Be that way!' if sentence.strip.empty?
    'Whatever.'
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.strip(input) == "" -> "Fine. Be that way!"
      String.ends_with?(input,"?") -> "Sure."
      String.upcase(input) == input && !String.match?(input,~r/^[0-9 ,]+$/) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end
